airflow/utils/log/bluecore_bq_gcs_task_handler.py

```diff
@@ -0,0 +1,194 @@
+                        container_name = string.replace(container['name'], '-', '_')
+                        # TODO: replace gke_logs with a variable
+                        bq_table_name = "{}.gke_logs.{}_{}".format(str(data_project_name), str(container_name),
+                                                                   str(bq_table_date))
+                        tables.append((str(bq_table_name), str(pod_id)))
+
+                query = ("\n UNION ALL \n ".join(
+                    [BQGCSTaskHandler.generate_query(bq_table_name=bq_table_name, pod_id=pod_id) for
+                     bq_table_name, pod_id in tables]) + "\n ORDER BY pod_id, logName, timestamp")
+
+                query_job = client.query(query)
+                result = query_job.result()
+
+                for row in result:
+                    log_lines.append(str("{}  {}".format(row.logName, row.textPayload)))
+
+                return "\n".join(log_lines)
+
+            # else statement from gcs_task_handler.py
+            else:
+                log = super(GCSTaskHandler, self)._read(ti, try_number)
+                return log
+
+    def get_pods(self, job_name):
+        """
+        Return pods for a given job_name
+
+        :param job_name: a unique job name
+        :type: string
+        :return: result form kubectl command that contains job pod and container information
+        :type: dict
+
+        """
+        return json.loads(BQGCSTaskHandler.retryable_check_output(args=[
+            'kubectl', 'get', 'pods', '-o', 'json', '-l', 'job-name==%s' % job_name]))
+
+    @staticmethod
+    def retryable_check_output(args, retry_count=3):
+        """
+        Reads the job description, retrying on failure
+        :param args: Arguments to pass to subprocess.check_output
+        :type args: List of string
+        :param retry_count: Number of times to retry (default=3)
+        :type retry_count: int
+        :return: string
+        """
+        try:
+            return subprocess.check_output(args=args)
+        except subprocess.CalledProcessError as e:
+            if retry_count > 0:
+                logging.info("Retrying check_output because %s" % e)
+                return BQGCSTaskHandler.retryable_check_output(args=args, retry_count=retry_count - 1)
+            else:
+                raise
+
+    @staticmethod
+    def generate_query(bq_table_name, pod_id):
+        """
+        Generates a query for the logs of a pod
+
+        :param bq_table_name: name of the table that will be queried
+        :type: string
+        :param pod_name: name of the pod that the logging information is derived from
+        :type: string
+        :return: query: the query for the specific pod logs
+        :type: string
+        """
+        query = (
+            """
+            SELECT logName, resource.labels.pod_id, timestamp, textPayload
+            FROM `{}`
+            WHERE resource.labels.pod_id = '{}'
```

Comment:
We want to order the log lines by timestamp, right?

Reply:
On line 98, I order them by timestamp. Since I union multiple queries into one, I add ORDER BY at the end

Reply:
:+1: gotcha